lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -95,28 +105,241 @@ class IntProgCmdProvider: public CmdProvider {
+        {
+            m_firmwareFile.close();
+        }
+
+        /*If the file is not yet open, open it.*/
+        m_firmwareFile = LittleFS.open(m_fileName, "r");
+        return false != m_firmwareFile;
 }
-#endif
+
+    /**
+     * @brief Retrieves the next command and the next response.
+     *
+     * This method returns the next command and its associated response
+     * from the command sequence memory.
+     * @param cmd Reference to a pointer to the next command.
+     * @param rsp Reference to a pointer to the next response.
+     * @return True if there is a next command and next response, false otherwise.
+     */
+    virtual bool next(const CommandInfo*& cmd, const ResponseInfo*& rsp)
+    {
+        if (m_nextCmd == FLSPRG_COMPLETE) 
+        {
+            cmd = nullptr;
+            rsp = nullptr;
+
+            if (m_firmwareFile)
+            {
+                m_firmwareFile.close(); // Close the file
+            }
+
+            return false;
+        }
+        else
+        {
+            rsp = &m_responses[m_nextCmd];
+    
+            /*Update the command with required content to write in the flash memory.*/
+            if(true == updateFwProgCmd(m_updatedCmd))
+            {
+                cmd = &m_updatedCmd;
+                return true;
+            }
+            else
+            {
+                cmd = nullptr;
+                rsp = nullptr;
+                return false;
+            }
+        }
+    }   
+
+private:
+    /**
+     * @brief Length of a flash memory block in bytes.
+     * This constant defines the length of a flash memory block in bytes.
+     * It is used to specify the size of the blocks when reading or writing data to flash memory.
+     */
+    static const size_t FLASH_BLOCK_LENGTH = 128U;
+
+    /**
+     * @brief Length of a write block command in bytes.
+     * This constant defines the length of a write block command in bytes.
+     * It is used to specify the size of the command for writing a block of data to memory.
+     */
+    static const size_t WRITE_BLOCK_CMD_LENGTH = 4U;
+
+    static  CommandInfo m_cmds[]; /**< Array of command information.*/
+    static  ResponseInfo m_responses[]; /**< Array of response information.*/
+    
+    CommandInfo m_updatedCmd; /**<Intermediate storage for updated commands.*/
+
+    const char* m_fileName = "/firmware.bin"; /**< Name of the file to be read.*/
+    uint8_t m_firmwareBuffer[128]; /**< Buffer to hold firmware data read from the file.*/
+    size_t m_firmwareBytesRead; /**< Number of bytes read from the firmware file.*/
+    File m_firmwareFile; /**< Handle for the firmware file.*/
+
+    /**
+     * @brief Enumeration of programming commands.
+     * This enumeration defines the various programming commands used during firmware programming.
+     * It includes commands for setting memory addresses and writing data blocks.
+     */
+   enum ProgCmds {
+    FLSPRG_SET_ADDR,    /**< Command to set the memory address. */

```

Comment:
indent wrong

Reply:
It will be taken care of by the next commit.